Snowflake.Data.Tests/IntegrationTests/SFConnectionPoolAsyncIT.cs

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+using Snowflake.Data.Tests.Util;
+
+namespace Snowflake.Data.Tests.IntegrationTests
+{
+    using System;
+    using System.Data;
+    using System.Data.Common;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+    using Moq;
+    using Snowflake.Data.Client;
+    using Snowflake.Data.Core;
+    using Snowflake.Data.Tests.Mock;
+
+    [TestFixture, NonParallelizable]
+    class SFConnectionPoolITAsync : SFBaseTestAsync
```

Comment:
Extracted class from file SFConnectionPoolT. Minor changes in couple of lines to remove IDE highlights